### PR TITLE
Vaxrank depends on the input BAM having a BAI

### LINKED ISSUE #356: Add dependence on .bai of bam file for Vaxrank

### PR DESCRIPTION
**Pull request by ihodes**
fixes #356

Question: will this fail if the BAM is unsorted? Does Vaxrank expect the BAM to be sorted? @iskandr

### CODE REVIEW

**Comment by arahuja** [comment]
@ihodes IIRC you shouldn't be able to index an unsorted BAM

**Comment by ihodes** [comment]
@arahuja Thanks! cool, that's what I thought, will add that in.

File: src/bfx_tools/vaxrank.ml [hunk 1]
```diff
@@ -172,12 +174,13 @@ let run ~(run_with: Machine.t)
...
+        depends_on (Samtools.index_to_bai ~run_with sorted_bam);
+        depends_on Machine.Tool.(ensure vaxrank);
+        depends_on (Pyensembl.cache_genome ~run_with ~reference_build);
+        depends_on bam;
+        depends_on alleles_file;
+      ] @ (List.map ~f:depends_on vcfs)
+        @ predictor_edges)
     ~make:(
       Machine.run_program run_with ~name
         Program.(
```
Comment (smondet) [inline src/bfx_tools/vaxrank.ml]: You also need to use `sorted_bam` in the `make` also (cf. `let arguments` above)

Comment (ihodes) [inline src/bfx_tools/vaxrank.ml]: Good catch, thank you

**Comment by ihodes** [comment]
@smondet fixed, back to you

**Pull request merged**